iceoryx_posh/source/roudi/service_registry.cpp

```diff
@@ -100,25 +100,25 @@ void ServiceRegistry::remove(const capro::ServiceDescription& serviceDescription
 }
 
 void ServiceRegistry::find(ServiceDescriptionVector_t& searchResult,
-                           const capro::IdString_t& service,
-                           const capro::IdString_t& instance) const noexcept
+                           const cxx::variant<cxx::Wildcard_t, capro::IdString_t>& service,
+                           const cxx::variant<cxx::Wildcard_t, capro::IdString_t>& instance) const noexcept
 {
     cxx::vector<uint64_t, MAX_SERVICE_DESCRIPTIONS> intersection;
 
     // Find (K1, K2)
     // O(log n + log n + max(#PossibleServices + #possiblesInstances) + #intersection)
-    if (instance != Wildcard && service != Wildcard)
+    if (instance.index() != 0 && service.index() != 0)
     {
         cxx::vector<uint64_t, MAX_SERVICE_DESCRIPTIONS> possibleServices;
         cxx::vector<uint64_t, MAX_SERVICE_DESCRIPTIONS> possibleInstances;
 
-        auto rangeServiceMap = m_serviceMap.equal_range(service);
+        auto rangeServiceMap = m_serviceMap.equal_range(*service.get_at_index<1>());
```

Comment:
I would not replace the index, but instead make the result of the call more explicit to convey intention.

I am not sure whether we can get a reference to the data contained in the `variant` (without copying), but if we can I would introduce a helper variable like `auto& serviceName = *service.get_at_index<1>();` and use this to make the code more expressive without any overhead (if it can be a reference the compiler will optimize it away).

If it is currently copied anyway it makes no difference performance-wise but will improve readability.

Reply:
Obsolete, because I replaced variant with optional.